internal/state/graph/gateway_listener.go

```diff
@@ -335,31 +335,47 @@ func createHTTPSListenerValidator(gwNsName string) listenerValidator {
 	}
 }
 
-func createHostnameConflictResolver() listenerConflictResolver {
-	usedHostnamesByPort := make(map[v1beta1.PortNumber]map[string]*Listener)
+func createPortConflictResolver() listenerConflictResolver {
+	conflictedPorts := make(map[v1beta1.PortNumber]bool)
+	portProtocolOwner := make(map[v1beta1.PortNumber]v1beta1.ProtocolType)
+	listenersByPort := make(map[v1beta1.PortNumber][]*Listener)
+
+	format := "Multiple listeners for the same port %d specify different protocols; " +
```

Comment:
we will probably support TLS and HTTPS on the same port, so better to say incompatible
```suggestion
	format := "Multiple listeners for the same port %d specify incompatible protocols; " +
```